File: Algorithm/simulator/Car.java
package simulator;

import java.awt.*;
import java.util.ArrayList;

public class Car{
	private int[] pos;
	private int facing;
	private int size; //car size will be size*size grid 
	public Car(int x,int y, int facing, int size){
		pos = new int[2];
		pos[0] = x;
		pos[1] = y;
		this.facing = facing;
		this.size = size;
	}
	public int[] getPos(){
		return pos;
	}
	
	public int getFacing(){
		return facing;
	}
	
	public int getSize(){
		return size;
	}
	
	//move forward and backwards: +ve = forward, -ve = backwards
	public void moveFB(int steps){
		switch(facing){
			//0-North, 2-East, 4-South, 6-West
			case 0:
				pos[1] -= steps;
				break;
			case 1:
				pos[0] += steps;
				pos[1] -= steps;
				break;
			case 2:
				pos[0] += steps;
				break;
			case 3:
				pos[0] += steps;
				pos[1] += steps;
				break;
			case 4:
				pos[1] += steps;
				break;
			case 5:
				pos[0] -= steps;
				pos[1] += steps;
				break;
			case 6:
				pos[0] -= steps;
				break;
			case 7:
				pos[0] -= steps;
				pos[1] -= steps;
				break;
			default:
				break;
		}
	}

	public void turn(int direction){
		//+ve clockwise, -ve anti-clockwise
		facing = (facing+direction)%8;
		if(facing<0)
			facing = 8 + facing;
	}
	
	
}

File: RPI/Car.py
import math

class Car:
    pos = [0,0]
    direction = 0
    size = 3
    def __init__(self,x,y,direction):
        self.pos = [x,y]
        self.direction = direction
        
    def moveFB(self, steps):
        #0=N, 2=E, 4=S, 6=W
        if self.direction==0:
            self.pos[1]-=steps
        elif self.direction ==1:
            self.pos[0]+=steps
            self.pos[1]-=steps
        elif self.direction ==2:
            self.pos[0]+=steps
        elif self.direction ==3:
            self.pos[0]+=steps
            self.pos[1]+=steps
        elif self.direction ==4:
            self.pos[1]+=steps
        elif self.direction ==5:
            self.pos[0]-=steps
            self.pos[1]+=steps
        elif self.direction ==6:
            self.pos[0]-=steps
        elif self.direction ==7:
            self.pos[0]-=steps
            self.pos[1]-=steps
            
    def turn(self, direction):
        self.direction = (self.direction+direction)%8
        if self.direction<0:
            self.direction = 8+self.direction
        #print(str(self.direction))
